enhancements/installer/connected-assisted-installer.md

```diff
@@ -0,0 +1,321 @@
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. Initial proposal
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+- Maturity levels - `Dev Preview`, `Tech Preview`, `GA`
+- Deprecation
+
+Clearly define what graduation means.
+
+#### Examples
+
+These are generalized examples to consider, in addition to the aforementioned
+[maturity levels][maturity-levels].
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+##### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+##### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+This work is all about building clusters on day 1. After the cluster
+is running, it should be possible to upgrade or downgrade it like any
+other cluster.
+
+### Version Skew Strategy
+
+The assisted installer and agent need to know enough about the
+installer version to construct its inputs correctly. This is a
+development-time skew, for the most part, and the service that builds
+the live ISOs with the assisted installer components should be able to
+adjust the version of the assisted installer to match the version of
+OpenShift, if necessary.
+
+## Implementation History
+
+### Proof of Concept (June, 2020)
+
+* https://github.com/filanov/bm-inventory : The REST service
+* https://github.com/ori-amizur/introspector : Gathers hardware and
+  connectivity info on a host
+* https://github.com/oshercc/coreos_installation_iso : Creates the
+  RHCOS ISO - runs as a k8s job by bm-inventory
+* https://github.com/oshercc/ignition-manifests-and-kubeconfig-generate :
+  Script that generates ignition manifests and kubeconfig - runs as a
+  k8s job by bm-inventory
+* https://github.com/tsorya/test-infra : Called by
+  openshift-metal3/dev-scripts to end up with a cluster of VMs like in
+  dev-scripts, but using the assisted installer
+* https://github.com/eranco74/assisted-installer.git : The actual
+  installer code that runs on the hosts
+
+## Drawbacks
+
+The idea is to find the best form of an argument why this enhancement should _not_ be implemented.
+
+## Alternatives
+
+The telco/edge bare metal team is working on support for automating
```

Comment:
```suggestion
The bare metal IPI team is working on support for automating
```

Reply:
We cannot say "IPI" in external docs.  We need to say "installer-provisioned infrastructure".  But do we really have a metal team that ignores user-provisioned infrastructure?

Reply:
Fixed in #382 

The team working on the automation does not support user-provisioned deployments today. IIUC, the main installer team supports those.